File: lintgen/generated_agent/qwen-full/qwen2.5:32b-instruct/base+docs_agent/BadHostnameVerifierDetector:BadHostnameVerifier/iter_7.java
package com.android.tools.lint.checks;

import com.android.annotations.NonNull;
import com.android.tools.lint.detector.api.Category;
import com.android.tools.lint.detector.api.Detector;
import com.android.tools.lint.detector.api.Implementation;
import com.android.tools.lint.detector.api.Issue;
import com.android.tools.lint.detector.api.JavaContext;
import com.android.tools.lint.detector.api.Scope;
import com.android.tools.lint.detector.api.Severity;
import org.jetbrains.uast.UCallExpression;
import org.jetbrains.uast.UClass;
import org.jetbrains.uast.UMethod;
import org.jetbrains.uast.UIastVisitor;
import org.jetbrains.uast.UReturnExpression;

import java.util.Collections;
import java.util.List;

public class BadHostnameVerifierDetector extends Detector implements Detector.UastScanner {
    private static final String HOSTNAME_VERIFIER = "javax.net.ssl.HostnameVerifier";
    private static final String VERIFY_METHOD_NAME = "verify";

    @Override
    public List<String> getApplicableMethodNames() {
        return Collections.singletonList(VERIFY_METHOD_NAME);
    }

    @Override
    public void visitMethod(@NonNull JavaContext context, @NonNull UMethod method) {
        if (HOSTNAME_VERIFIER.equals(method.getContainingClass().getQualifiedName())) {
            final boolean[] alwaysReturnsTrue = {false};
            method.accept(new UIastVisitor() {
                @Override
                public boolean visitReturnExpression(UReturnExpression node) {
                    if ("true".equals(node.getValue().getText())) {
                        alwaysReturnsTrue[0] = true;
                    }
                    return super.visitReturnExpression(node);
                }
            });

            if (alwaysReturnsTrue[0]) {
                context.report(ISSUE, method, context.getLocation(method),
                        "Insecure HostnameVerifier implementation that always returns true");
            }
        }
    }

    private static final Issue ISSUE = Issue.create(
            "BadHostnameVerifier",
            "Insecure HostnameVerifier implementation that always returns true",
            "Implementations of `HostnameVerifier` whose `verify` method always returns true could result in insecure network traffic caused by trusting arbitrary hostnames in TLS/SSL certificates presented by peers.",
            Category.SECURITY,
            5,
            Severity.WARNING,
            new Implementation(
                    BadHostnameVerifierDetector.class,
                    Scope.JAVA_FILE_SCOPE
            )
    );
}

File: lintgen/generated_agent/qwen-full/qwen2.5:32b-instruct/base+apis+docs_agent/EmptySuperDetector:EmptySuperCall/iter_7.kt
package com.android.tools.lint.checks

import com.android.tools.lint.detector.api.*
import org.jetbrains.uast.*

class EmptySuperDetector : Detector(), SourceCodeScanner {

    companion object {
        val ISSUE = Issue.create(
            id = "EmptySuper",
            briefDescription = "Calling an empty super method",
            explanation = """
                Methods annotated with `@EmptySuper` should not call the super implementation in overriding methods, either because it is empty or contains code not intended to be run when overridden.
            """,
            category = Category.CORRECTNESS,
            priority = 6,
            severity = Severity.WARNING,
            implementation = Implementation(
                EmptySuperDetector::class.java,
                Scope.JAVA_FILE_SCOPE
            )
        )
    }

    override fun getApplicableMethodNames(): List<String>? {
        return null
    }

    private fun isAnnotatedWithEmptySuper(element: UElement): Boolean {
        return element.annotations.any { it.qualifiedName == "com.android.tools.lint.detector.api.EmptySuper" }
    }

    private fun checkForSuperCall(context: JavaContext, method: UMethod) {
        val body = method.bodyExpression as? UBlockExpression ?: return
        for (expr in body.expressions) {
            if (expr is UCallExpression && expr.methodName == "super") {
                context.report(
                    ISSUE,
                    expr,
                    context.getLocation(expr),
                    "Overriding methods should not call super when annotated with @EmptySuper"
                )
            }
        }
    }

    override fun getApplicableUastTypes(): List<Class<out UElement>>? {
        return listOf(UMethod::class.java)
    }

    override fun createUastHandler(context: JavaContext): UElementHandler? {
        return object : UElementHandler() {
            override fun visitMethod(node: UMethod) {
                if (isAnnotatedWithEmptySuper(node)) {
                    checkForSuperCall(context, node)
                }
            }
        }
    }

}